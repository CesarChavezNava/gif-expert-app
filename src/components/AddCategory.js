import React, { useState } from 'react'
import PropTypes from 'prop-types';

export const AddCategory = ({setCategories}) => {
    const initialState = '';
    const [inputValue, setInputValue] = useState(initialState);

    const handleInputChange = (e) => {
        setInputValue(e.target.value);
    }

    const handleSubmit = (e) => {
        e.preventDefault();

        if(inputValue.trim().length > 2) {
            setCategories( categories => [...categories, inputValue]);
            setInputValue('');
        }
    }

    return (
        <form onSubmit={handleSubmit}>
            <input type="text" value={inputValue} onChange={handleInputChange} />
            <h2>Add Category</h2>
        </form>
    )
}

AddCategory.prototype = {
    setCategories: PropTypes.func.isRequired
}
