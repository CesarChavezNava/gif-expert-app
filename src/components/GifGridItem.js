import React from 'react'

export const GifGridItem = ( { title, url} ) => {
    return (
        <div className="column is-4">
            <div className="card">
                <div className="card-image">
                    <figure className="image is-4by3">
                        <img src={url} alt={title} />
                    </figure>
                </div>
                <div className="card-content">
                    <div className="content">
                        <p>{title}</p>
                    </div>
                </div>
            </div>
        </div>
    )
}
