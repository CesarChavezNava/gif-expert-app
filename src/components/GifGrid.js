import React from 'react'
import PropTypes from 'prop-types';

import { GifGridItem } from './GifGridItem';
import { useFetchGifs } from '../hooks/useFetchGifs';

export const GifGrid = ({ category }) => {

    const { data: images, loading } = useFetchGifs(category);

    return (
        <div className="container is-fluid">
            { loading && <progress className="progress is-small is-primary" max="100">15%</progress> }
           <h3 className="subtitle">{category}</h3> 
           <div className="columns">
               {
                   images.map((image) => (
                    <GifGridItem key={image.id} {...image} />
                   ))
               }
           </div>
        </div>
    )
}

GifGrid.prototype = {
    category: PropTypes.string.isRequired
}