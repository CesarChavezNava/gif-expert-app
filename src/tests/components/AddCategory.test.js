import React from 'react';
import '@testing-library/jest-dom';
import { shallow } from 'enzyme';

import { AddCategory } from '../../components/AddCategory';

describe('Pruebas del componente AddCategory', () => {
    const setCategories = jest.fn();
    let wrapper = shallow(<AddCategory setCategories={setCategories} />);

    beforeEach(() => {
        jest.clearAllMocks();
        wrapper = shallow(<AddCategory setCategories={setCategories} />);
    })

    test('Debe mostrar el componente correctamente ', () => {    
        expect(wrapper).toMatchSnapshot();
    });

    test('Debe de cambiar la caja de texto', () => {
        const value = 'Overwatch';
        const input = wrapper.find('input');
        input.simulate('change', { target: { value: value } });
    });

    test('No debe de postear la informaciÃ³n con submit ', () => {
        const form = wrapper.find('form');
        form.simulate('submit', { preventDefault(){} });

        expect(setCategories).not.toHaveBeenCalled();
    });
    
    test('Debe de llamar el setCategories y limpiar la caja de texto', () => {
        const input = wrapper.find('input');
        input.simulate('change', { target: { value: 'Overwatch' } });

        const form = wrapper.find('form');
        form.simulate('submit', { preventDefault(){} });

        expect(setCategories).toHaveBeenCalled();
        expect(setCategories).toHaveBeenCalledWith(expect.any(Function));
        expect(input.prop('value')).toBe('');
    })
    
    
});
