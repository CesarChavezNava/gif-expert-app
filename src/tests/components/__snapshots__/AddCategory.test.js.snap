// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas del componente AddCategory Debe mostrar el componente correctamente  1`] = `
<form
  onSubmit={[Function]}
>
  <input
    className="input is-primary"
    onChange={[Function]}
    type="text"
    value=""
  />
</form>
`;
